pkg/ravencontroller/controllers/gateway_controller.go

```diff
@@ -155,7 +155,11 @@ func (r *GatewayReconciler) electActiveEndpoint(nodeList corev1.NodeList, gw *ra
 
 	// the current active endpoint is still competent.
 	if checkActive(gw.Status.ActiveEndpoint) {
-		return gw.Status.ActiveEndpoint.DeepCopy()
+		for _, v := range gw.Spec.Endpoints {
```

Comment:
I think `checkActive` has already assured that `gw.Status.ActiveEndpoint` is in the `gw.Spec.Endpoints` list.  Did you find any bugs with the original code?

Reply:
After the **controller manager** elects the `ActiveEndpoint`, **Raven** will update the `publicIP` of the `Active Endpoint`，so we should use the copy of updated one.

Reply:
Got it.